TF-9289 document priority varsets (#458)
hashicorp/terraform-docs-common: website/docs/cloud-docs/workspaces/variables/managing-variables.mdx

Variables within a variable set can also automatically overwrite variables with the same key in other variable sets applied to the same workspace. Though variable sets are created for the organization, these overwrites occur within each workspace. Refer to [variable precedence](/terraform/cloud-docs/workspaces/variables#precedence) for more details.

## Priority Variable Sets

You can select to prioritize variable values from a variable set over any other variables with the same key set at more specific scopes. Variables from prioritized variable sets will also overwrite variables with the same key that are set from the command line, .*auto.tfvars files, and terraform.tfvars files.

The only variable values that a prioritized variable set cannot overwrite are values set directly in the terraform configuration itself. If you want to fully ensure the values of variables including values in your terraform configuration code, we recommend using a Policy Check or Run Task.

> 

Author: ```suggestion
Refer to [variable precedence](/terraform/cloud-docs/workspaces/variables#precedence-with-priority-variable-sets) for more details.

```

Variables within a variable set can also automatically overwrite variables with the same key in other variable sets applied to the same workspace. Though variable sets are created for the organization, these overwrites occur within each workspace. Refer to [variable precedence](/terraform/cloud-docs/workspaces/variables#precedence) for more details.

## Priority Variable Sets

> You can select to prioritize variable values from a variable set over any other variables with the same key set at more specific scopes. Variables from prioritized variable sets will also overwrite variables with the same key that are set from the command line, .*auto.tfvars files, and terraform.tfvars files.

Author: ```suggestion
You can select to prioritize variable values from a variable set over any other variables with the same key set at more specific scopes. Variables from prioritized variable sets will also overwrite variables with the same key that are set from the command line, `.*auto.tfvars` files, and `terraform.tfvars` files.
```

> It is still possible for a user to directly modify the Terraform configuration and remove usage of a variable and replace it with a hard coded value. If you are looking for stricter enforcement than this, we recommend using policy checks or run tasks.

Reviewer: ```suggestion
It is still possible for a user to directly modify the Terraform configuration and remove usage of a variable and replace it with a hard coded value. For stricter enforcement, we recommend using policy checks or run tasks.
```

just to avoid passive voice!

> The only variable values that a prioritized variable set cannot overwrite are variables with values set directly in the terraform configuration itself. If you want to fully ensure the values of variables including values in your terraform configuration code, we recommend using a Policy Check or Run Task.

Author: ```suggestion
It is still possible for a user to directly modify the Terraform configuration and remove usage of a variable and replace it with a hard coded value. If you are looking for stricter enforcement than this, we recommend using policy checks or run tasks.
```

Author: @ritsok what about re-using what Sarah suggested down here as well. Hopefully that also answers your question below.

Reviewer: suggested a tweak!

> The only variable values that a prioritized variable set cannot overwrite are variables with values set directly in the terraform configuration itself. If you want to fully ensure the values of variables including values in your terraform configuration code, we recommend using a Policy Check or Run Task.

Reviewer: ```suggestion
While a priority variable set can enforce specific variable values, it does not guarantee that the configuration actually references that variable instead of hardcoding a value. For stricter enforcement mechanisms, we recommend using policy checks or run tasks.
```
